mii/api.py

```diff
@@ -0,0 +1,149 @@
+
+    # Fill model_config dict with relevant kwargs, store remaining kwargs in a new dict
+    remaining_kwargs = {}
+    for key, val in kwargs.items():
+        if key in ModelConfig.__dict__["__fields__"]:
+            if key in model_config:
+                assert model_config.get(key) == val, f"{key} in model_config must match {key}"
+            model_config[key] = val
+        else:
+            remaining_kwargs[key] = val
+
+    # Create the ModelConfig object and return it with remaining kwargs
+    model_config = ModelConfig(**model_config)
+    return model_config, remaining_kwargs
+
+
+def _parse_kwargs_to_mii_config(model_name_or_path: str = "",
+                                model_config: Optional[Dict[str,
+                                                            Any]] = None,
+                                mii_config: Optional[Dict[str,
+                                                          Any]] = None,
+                                **kwargs) -> MIIConfig:
+    # Parse all model_config kwargs
+    model_config, remaining_kwargs = _parse_kwargs_to_model_config(model_name_or_path=model_name_or_path, model_config=model_config, **kwargs)
+
+    if mii_config is None:
+        mii_config = {}
+
+    assert isinstance(mii_config, dict), "mii_config must be a dict"
+
+    # Verify that any model_config kwargs match any existing model_config in the mii_config
+    if "model_config" in mii_config:
+        assert mii_config.get("model_config") == model_config, "mii_config['model_config'] must match model_config"
+    else:
+        mii_config["model_config"] = model_config
+
+    # Fill mii_config dict with relevant kwargs, raise error on unknown kwargs
+    for key, val in remaining_kwargs.items():
+        if key in MIIConfig.__dict__["__fields__"]:
+            if key in mii_config:
+                assert mii_config.get(key) == val, f"{key} in mii_config must match {key}"
+            mii_config[key] = val
+        else:
+            raise UnknownArgument(f"Keyword argument {key} not recognized")
+
+    # Return the MIIConfig object
+    mii_config = MIIConfig(**mii_config)
+    return mii_config
+
+
+def client(model_or_deployment_name: str) -> MIIClient:
+    mii_config = get_mii_config(model_or_deployment_name)
+
+    return MIIClient(mii_config)
+
+
+def serve(model_name_or_path: str = "",
+          model_config: Optional[Dict[str,
+                                      Any]] = None,
+          mii_config: Optional[Dict[str,
+                                    Any]] = None,
+          **kwargs) -> Union[None,
+                             MIIClient]:
+    mii_config = _parse_kwargs_to_mii_config(model_name_or_path=model_name_or_path,
+                                             model_config=model_config,
+                                             mii_config=mii_config,
+                                             **kwargs)
+
+    #MIIServer(mii_config)
```

Comment:
Can we remove this?

Reply:
Let me update this explain why I'm keeping it here!